assets/js/modules/analytics/datastore/settings.js

```diff
@@ -226,3 +231,47 @@ export function validateCanSubmitChanges( select ) {
 		select( MODULES_ANALYTICS_4 ).__dangerousCanSubmitChanges();
 	}
 }
+
+/**
+ * Gets the value of canUseSnippet based on the gaPropertyID of tagmanager module and propertyID.
+ *
+ * @since n.e.x.t
+ *
+ * @return {boolean|undefined} Computed value of canUseSnippet. `undefined` if not loaded.
+ */
+export const getCanUseSnippet = createRegistrySelector( ( select ) => () => {
+	const analyticsSettings = select( MODULES_ANALYTICS ).getSettings();
+
+	const isTagManagerConnected = select( CORE_MODULES ).isModuleConnected(
+		'tagmanager'
+	);
+
+	if (
+		isTagManagerConnected === undefined ||
+		analyticsSettings === undefined
+	) {
+		return undefined;
+	}
+
+	if ( isTagManagerConnected === null || isTagManagerConnected === false ) {
+		return analyticsSettings.canUseSnippet;
+	}
+
+	const tagManagerUseSnippet = select( MODULES_TAGMANAGER ).getUseSnippet();
```

Comment:
Before calling any selector in this store, we should double-check whether it is loaded - which should be the case given the module is connected, but it's still a better practice to first check that `select( MODULES_TAGMANAGER )` is not `null` (also mentioned in the ACs).